L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -5,47 +5,93 @@
 
 HGCalConcentratorSuperTriggerCellImpl::
 HGCalConcentratorSuperTriggerCellImpl(const edm::ParameterSet& conf)
-  : stcSize_(conf.getParameter< std::vector<unsigned> >("stcSize"))
+  : stcSize_(conf.getParameter< std::vector<unsigned> >("stcSize")),
+    fixedDataSize_(conf.getParameter<bool>("fixedDataSize"))
 {
 
     if ( stcSize_.size() != kNLayers_ ){
         throw cms::Exception("HGCTriggerParameterError")
             << "Inconsistent size of super trigger cell size vector" << stcSize_.size() ;
     }
     for(auto stc : stcSize_) {
-        if ( stc!=kSTCsizeFine_ && stc!=kSTCsizeCoarse_ ){
+        if ( stc!=kSTCsizeFine_ && stc!=kSTCsizeCoarse_ && stc!=kSTCsizeMid_ && stc!=kSTCsizeVeryFine_){
             throw cms::Exception("HGCTriggerParameterError")
               << "Super Trigger Cell should be of size "<<
-              kSTCsizeFine_ << " or " << kSTCsizeCoarse_;
+              kSTCsizeVeryFine_ << " or " << kSTCsizeFine_ <<
+              kSTCsizeMid_ << " or " << kSTCsizeCoarse_;
         }
     }
+
+    std::string energyType_(conf.getParameter<string>("type_energy_division"));
+
+    if(energyType_=="superTriggerCell"){
+      energyDivisionType_ = superTriggerCell;
+    }else if(energyType_=="oneBitFraction"){
+      energyDivisionType_ = oneBitFraction;
+    }else if(energyType_=="equalShare"){
+      energyDivisionType_ = equalShare;
+    }else if(energyType_=="coarse2TriggerCell"){
+      energyDivisionType_ = coarse2TriggerCell;
+    }else {
+      energyDivisionType_ = superTriggerCell;
+    } 
+
     
 }
 
 std::map<int,int> 
 HGCalConcentratorSuperTriggerCellImpl::kSplit_ = {
+  {kSTCsizeVeryFine_, kSplit_v8_VeryFine_},
   {kSTCsizeFine_, kSplit_v8_Fine_},
+  {kSTCsizeMid_, kSplit_v8_Mid_},
   {kSTCsizeCoarse_, kSplit_v8_Coarse_}
 };
 
+std::map<int,int> 
+HGCalConcentratorSuperTriggerCellImpl::kSplit_v9_ = {
+  {kSTCsizeVeryFine_, kSplit_v9_VeryFine_},
+  {kSTCsizeFine_, kSplit_v9_Fine_},
+  {kSTCsizeMid_, kSplit_v9_Mid_},
+};
+
 int
-HGCalConcentratorSuperTriggerCellImpl::getSuperTriggerCellId(int detid) const {
+HGCalConcentratorSuperTriggerCellImpl::getSuperTriggerCellId(int detid, int STCsize) const {
```

Comment:
I suppose that this function should depend whether a TC coarsening is done before or not because TC IDs are redefined in case of coarsening. But it doesn't seem to be implemented.

Reply:
The coarsened TC ids are set to take the even ID, even if they are odd. They will still end up in the correct superTC regardless (at least for v8 I think).